packages/get-platform/src/getPlatform.ts

```diff
@@ -54,64 +70,149 @@ export async function getos(): Promise<GetOSResult> {
+   * Ubuntu       => ID=ubuntu, ID_LIKE=debian                     => targetDistro=debian, familyDistro=debian
+   * Arch Linux   => ID=arch                                       => targetDistro=debian, familyDistro=arch
+   * Manjaro      => ID=manjaro, ID_LIKE=arch                      => targetDistro=debian, familyDistro=arch
+   * Red Hat      => ID=rhel, ID_LIKE=fedora                       => targetDistro=rhel, familyDistro=rhel
+   * Centos       => ID=centos, ID_LIKE=rhel fedora                => targetDistro=rhel, familyDistro=rhel
+   * Alma Linux   => ID="almalinux", ID_LIKE="rhel centos fedora"  => targetDistro=rhel, familyDistro=rhel
+   * Fedora       => ID=fedora                                     => targetDistro=rhel, familyDistro=rhel
+   */
 
-  if (id === 'nixos') {
-    return 'nixos'
-  }
+  const distroInfo = match(id)
+    .with(
+      'alpine',
+      (originalDistro) =>
+        ({
+          targetDistro: 'musl',
+          familyDistro: originalDistro,
+          originalDistro,
+        } as const),
+    )
+    .with(
+      'raspbian',
+      (originalDistro) =>
+        ({
+          targetDistro: 'arm',
+          familyDistro: 'debian',
+          originalDistro,
+        } as const),
+    )
+    .with(
+      'nixos',
+      (originalDistro) =>
+        ({
+          targetDistro: 'nixos',
+          originalDistro,
+          familyDistro: 'nixos',
+        } as const),
+    )
+    .with(
+      'debian',
+      'ubuntu',
+      (originalDistro) =>
+        ({
+          targetDistro: 'debian',
+          familyDistro: 'debian',
+          originalDistro,
+        } as const),
+    )
+    .with(
+      'rhel',
+      'centos',
+      'fedora',
+      (originalDistro) =>
+        ({
+          targetDistro: 'rhel',
+          familyDistro: 'rhel',
+          originalDistro,
+        } as const),
+    )
+    .otherwise((originalDistro) => {
+      if (idLike.includes('debian') || idLike.includes('ubuntu')) {
+        return {
+          targetDistro: 'debian',
+          familyDistro: 'debian',
+          originalDistro,
+        } as const
+      }
 
-  if (idLike.includes('centos') || idLike.includes('fedora') || idLike.includes('rhel') || id === 'fedora') {
-    return 'rhel'
-  }
+      if (id === 'arch' || idLike.includes('arch')) {
+        return {
+          targetDistro: 'debian',
+          familyDistro: 'arch',
+          originalDistro,
+        } as const
+      }
 
-  if (idLike.includes('debian') || idLike.includes('ubuntu') || id === 'debian') {
-    return 'debian'
-  }
+      if (idLike.includes('centos') || idLike.includes('fedora') || idLike.includes('rhel')) {
+        return {
+          targetDistro: 'rhel',
+          familyDistro: 'rhel',
+          originalDistro,
+        } as const
+      }
+
+      /* Generic distro info fallback */
+      return {
+        targetDistro: undefined,
```

Comment:
Personal preference: I would prefer that 'debian' is explicitly the default here because it's not obvious reading the code. Totally optional, as it is not the point of your PR.

Reply:
The default (`debian`) is explicitly spelled out in `getPlatform` [here](https://github.com/prisma/prisma/blob/e697f0b099c59b0e313ee15aa404fe9730718691/packages/get-platform/src/getPlatform.ts#L394-L402), where also custom warnings are defined when `targetDistro === undefined`